### PR TITLE
gpsd: fix: Allow starting as a service, and caveats about its use.

### PR DESCRIPTION
**Pull request by micolous**
This adds a plist which allows `gpsd` to automatically start, and adds some caveats about using it this way.

It looks like USB autodetection on Homebrew's `gpsd` binary is broken, and there doesn't look like an easy way to have user-specified configuration on plists from Homebrew (an analog to `/etc/default/gpsd` on Linux distros).  I've worked around this by:

- Having the daemon start up and opening its control socket.
- Giving instructions on how to manually add a device using this control socket.

I've also added instructions for starting up the daemon manually, which allows it to work with a user-specified device node at startup time.

This doesn't include any code patches to `gpsd` itself, and starting as a daemon is only offered as a convenience and is not mandatory. As a result, there's no changes to the binary or its checksum.

It also adds an extra note that `gpsd` doesn't integrate with macOS Location Services (CoreLocation).

### CODE REVIEW

File: Formula/gpsd.rb [hunk 1]
```diff
@@ -19,4 +19,70 @@ def install
...
+      <string>#{var}/log/gpsd.log</string>
+      <key>StandardErrorPath</key>
+      <string>#{var}/log/gpsd.log</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    assert_equal "#{opt_sbin}/gpsd: 3.16 (revision 3.16)", shell_output("#{sbin}/gpsd -V").chomp
```
Comment (fxcoudert) [inline Formula/gpsd.rb]:
```
Error: gpsd: failed
<"/usr/local/opt/gpsd/sbin/gpsd: 3.16 (revision 3.16)"> expected but was
<"/usr/local/Cellar/gpsd/3.16/sbin/gpsd: 3.16 (revision 3.16)">.
```

**Review by fxcoudert — DISMISSED** [review]
Would USB auto detection work better if we made the `libusb` dependency required?

**Comment by micolous** [comment]
Regardless of whether USB autodetection worked on OSX, you would still need
the control socket configured.

File: Formula/gpsd.rb (hunk 1 above)
Comment (micolous) [reply Formula/gpsd.rb]: Seems your bot is silly, and doesn't handle emails properly -- this is still waiting response from you.

Regardless of whether USB autodetection worked on OSX, you would still need
the control socket configured.

**Comment by ilovezfs** [comment]
Going to pass on this as this is the purview of upstream documentation, and the test isn't working, but thanks nonetheless for the PR!

**Comment by micolous** [comment]
This isn't a documentation PR, and setting up the plist is _only_ possible with the package manager, and documentation about this setup is well within the purview of Homebrew.

It's not clear to me that you actually use this package -- the out-of-the-box configuration Homebrew delivers is basically unusable.

**Comment by micolous** [comment]
I'll have another look at that test, as I thought I fixed that one.  I'm trying to dig up the code for it again.

**Comment by ilovezfs** [comment]
@micolous can you try to make the caveat shorter please? Its length is way beyond what we'd usually consider putting in any formula.

**Comment by micolous** [comment]
Done. The failing tests look like there's an issue with binaries being served for `scons`:

```
==> Downloading https://homebrew.bintray.com/bottles/scons-3.0.0_1.el_capitan.bottle.tar.gz
curl: (22) The requested URL returned error: 404 Not Found
==> Retrying download
==> Downloading https://homebrew.bintray.com/bottles/scons-3.0.0_1.el_capitan.bottle.tar.gz
curl: (22) The requested URL returned error: 404 Not Found
Error: Failed to download resource "scons"
Download failed: https://homebrew.bintray.com/bottles/scons-3.0.0_1.el_capitan.bottle.tar.gz

```

**Comment by ilovezfs** [comment]
Yes all of Bintray is down. I just contacted them a few minutes ago.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @micolous! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**